### ISSUE TITLE
Why mount minio instance to flask_app_ctx

### ISSUE THREAD

**Issue by DDusky** [alejandroandreu/flask-minio#3]
I don't understand the purpose of binding minio instances to flask application context， i hope the author can solve my doubts

```
    def connection(self):
        ctx = _app_ctx_stack.top
        if ctx is not None:
            if not hasattr(ctx, 'minio'):
                ctx.minio = self.connect()
            return ctx.minio

```
just have a problem: every request will call connect.
Thanks

**Comment by hiven** [comment]
No, the connection() method you've shared will not be automatically invoked with every HTTP request. It's a method you'd typically call explicitly in your code when you need to interact with MinIO or establish a connection to it.

However, if you're using this method within a route or function that handles an HTTP request and it's called within that route's execution context, then yes, it will be invoked when that specific route or function is triggered by an HTTP request.

The purpose of storing the MinIO connection in the application context is to reuse the connection across multiple parts of your application within the same context to avoid establishing multiple connections unnecessarily. This minimizes the overhead of creating new connections for each component that needs to interact with MinIO within that context.

Would like the author to confirm though!
